### ISSUE TITLE
mkdir(): File exists

### ISSUE THREAD

**Issue by UFOMelkor** [phpmetrics/PhpMetrics#240]
Hi,

This issue occured:

mkdir(): File exists

**Environment**

+ PHP: 7.0.9
+ PhpMetrics: v2.0.0
+ Operating System: Linux bgfa854 4.7.0-2-default #1 SMP PREEMPT Thu Aug 4 12:44:37 UTC 2016 (df207c0) x86_64

**Backtrace**

+ File: /home/bley/PhpstormProjects/phpmetrics/src/functions.php (line 93)
+ /home/bley/PhpstormProjects/phpmetrics/src/functions.php (line 93)
+ /home/bley/PhpstormProjects/phpmetrics/src/Hal/Report/Html/Reporter.php (line 63)
+ /home/bley/PhpstormProjects/phpmetrics/src/Hal/Application/Application.php (line 71)
+ /home/bley/PhpstormProjects/phpmetrics/bin/phpmetrics (line 20)

**Debug**

```
filename: src/functions.php
statements: /**
 * Class MyVisitor
 */
class MyVisitor extends \PhpParser\NodeVisitorAbstract
{
    /**
     * @var
     */
    private $callback;
    /**
     * MyVisitor constructor.
     * @param $callback
     */
    public function __construct($callback)
    {
        $this->callback = $callback;
    }
    /**
     * @inheritdoc
     */
    public function leaveNode(\PhpParser\Node $node)
    {
        call_user_func($this->callback, $node);
    }
}
/**
 * @param $node
 * @param $callback
 */
function iterate_over_node($node, $callback)
{
    $myVisitor = new \MyVisitor($callback);
    $traverser = new \PhpParser\NodeTraverser();
    $traverser->addVisitor($myVisitor);
    $traverser->traverse(array($node));
    return;
}
/**
 * @param $node
 * @return string
 */
function getNameOfNode($node)
{
    if (is_string($node)) {
        return $node;
    }
    if ($node instanceof \PhpParser\Node\Name\FullyQualified) {
        return (string) $node;
    }
    if ($node instanceof \PhpParser\Node\Expr\New_) {
        return getNameOfNode($node->class);
    }
    if (isset($node->class)) {
        return getNameOfNode($node->class);
    }
    if ($node instanceof \PhpParser\Node\Name) {
        return (string) implode($node->parts);
    }
    if ($node->name instanceof \PhpParser\Node\Expr\Variable) {
        return (string) $node->name->name;
    }
    if ($node->name instanceof \PhpParser\Node\Expr\MethodCall) {
        return getNameOfNode($node->name);
    }
    if ($node instanceof \PhpParser\Node\Expr\ArrayDimFetch) {
        return getNameOfNode($node->var);
    }
    if ($node->name instanceof \PhpParser\Node\Expr\BinaryOp) {
        return get_class($node->name);
    }
    return (string) $node->name;
}
/**
 * @param $src
 * @param $dst
 */
function recurse_copy($src, $dst)
{
    $dir = opendir($src);
    @mkdir($dst);
    while (false !== ($file = readdir($dir))) {
        if ($file != '.' && $file != '..') {
            if (is_dir($src . '/' . $file)) {
                recurse_copy($src . '/' . $file, $dst . '/' . $file);
            } else {
                copy($src . '/' . $file, $dst . '/' . $file);
            }
        }
    }
    closedir($dir);
}
/**
 * @return string
 */
function getVersion()
{
    return 'v2.0.0';
}

```

If I disable the issuer, it works without an error … :-/

**Comment by Halleck45** [comment]
> If I disable the issuer, it works without an error … :-/

Argh ! I'll fix it :cry:

**Comment by Halleck45** [comment]
@UFOMelkor I'm not really proud of the `Issuer`. But we need to have lot of informations (parsed code, PHP version...) when someone reports an issue.

Do you have a better idea ?

**Comment by Halleck45** [comment]
Closed by #246
